src/openfluid/ui/waresdev/SignalMigrationListener.hpp

```diff
@@ -0,0 +1,255 @@
+/*
+
+  This file is part of OpenFLUID software
+  Copyright(c) 2007, INRA - Montpellier SupAgro
+
+
+ == GNU General Public License Usage ==
+
+  OpenFLUID is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OpenFLUID is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OpenFLUID. If not, see <http://www.gnu.org/licenses/>.
+
+
+ == Other Usage ==
+
+  Other Usage means a use of OpenFLUID that is inconsistent with the GPL
+  license, and requires a written agreement between You and INRA.
+  Licensees for Other Usage of OpenFLUID may use this file in accordance
+  with the terms contained in the written agreement between You and INRA.
+  
+*/
+
+
+/**
+  @file SignalMigrationListener.hpp
+
+  @author Jean-Christophe FABRE <jean-christophe.fabre@inra.fr>
+*/
+
+
+#ifndef __OPENFLUID_UIWARESDEV_SIGNALMIGRATIONLISTENER_HPP__
+#define __OPENFLUID_UIWARESDEV_SIGNALMIGRATIONLISTENER_HPP__
+
+
+#include <iostream>
+#include <iomanip>
+
+#include <openfluid/waresdev/WareSrcMigratorListener.hpp>
+#include <openfluid/tools/Console.hpp>
+
+
+namespace openfluid { namespace ui { namespace waresdev {
+
+
+class SignalMigrationListener : public QObject, public openfluid::waresdev::WareSrcMigratorListener
```

Comment:
Is this class implementation consistent with similar classes in DevStudio?

Reply:
This class could be compared to `RunSimulationListener`, current implementation shows significant differences, see if could be more consistent